go/vt/schemadiff/table.go

```diff
@@ -1111,37 +1135,41 @@ func (c *CreateTableEntity) diffOptions(alterTable *sqlparser.AlterTable,
 						}
 						if option2AutoIncrement > option1AutoIncrement {
 							// never decrease AUTO_INCREMENT. Only increase
-							alterTableOptions = append(alterTableOptions, t2Option)
+							modifyTableOption(t1Option, t2Option)
 						}
 					case AutoIncrementIgnore:
 						// do not apply
 					}
 				default:
 					// Apply the new options
-					alterTableOptions = append(alterTableOptions, t2Option)
+					modifyTableOption(t1Option, t2Option)
```

Comment:
Seems like it's probably relatively easy to cover the default in the unit test too?

Reply:
Added.